sdcm/nemesis.py

```diff
@@ -2524,22 +2524,15 @@ def _corrupt_data_file(self):
                 'sudo dd if=/dev/urandom of={} count=1024'.format(sstable_file))
             self.log.debug('File {} was corrupted by dd'.format(sstable_file))
 
-    def disable_disrupt_corrupt_then_scrub(self):
-
-        # Flush data to sstable
-        self.target_node.run_nodetool("flush")
-
-        self.log.debug("Randomly corrupt sstable file")
-        self._corrupt_data_file()
-
-        self.log.debug("Rebuild sstable file by scrub, corrupted data file will be skipped.")
-        for ks_cf in self.cluster.get_non_system_ks_cf_list(self.target_node):
-            self.target_node.run_nodetool("scrub", args=f"{ks_cf.split('.')[0]}")
-
-        self.log.debug(
-            'Refreshing the cache by restart the node, and verify the rebuild sstable can be loaded successfully.')
-        self.target_node.stop_scylla_server(verify_up=False, verify_down=True)
-        self.target_node.start_scylla_server(verify_up=True, verify_down=False)
+    def disrupt_corrupt_then_scrub(self):
+        """
+        Try to rebuild the sstables of all test keyspaces by scrub, the corrupted partitions
+        will be skipped.
+        """
+        self.log.debug("Rebuild sstables by scrub with `--skip-corrupted`, corrupted partitions will be skipped.")
+        with ignore_scrub_invalid_errors():
```

Comment:
But if we have this filter we won't be able to see issues with sstables, right?

Reply:
Here we only ignored two kinds of errors, it won't always happen. But it's expected/acceptable if it occurs.

* "Skipping invalid clustering row fragment",
* "Skipping invalid partition",

The other errors can still send event and failed the nemesis (or break the test).